### PR TITLE
[fix] edge case in `MaxPool1d` and add ErrorInputs

### LINKED ISSUE #83224: parameter 'padding' of torch.nn.MaxPool1d can be set to negative integral
### 🐛 Describe the bug

Parameter 'padding' of torch.nn.MaxPool1d must be >= 0 and <= kernel_size / 2 in documentation. However, I found that negative integral can be set to torch.nn.MaxPool2d.

```
import torch
results={}
arg_1 = 2
arg_2 = 50
arg_3 = True
arg_class = torch.nn.MaxPool1d(arg_1,padding=-1,stride=arg_2,return_indices=arg_3,)
arg_4 = torch.rand([0, 1, 49], dtype=torch.float32)
results['res'] = arg_class(arg_4)
```
Above code works.

### Versions

pytorch: 1.8.1
python: 3.8.3
os: win11

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

### PR DESCRIPTION
**Pull request by khushi-411**
Fixes #83224

cc @kshitij12345 @albanD!

### CODE REVIEW

File: aten/src/ATen/native/Pool.h [hunk 1]
```diff
@@ -58,6 +58,8 @@ template<typename T>
 static inline T pooling_output_shape(
       T inputSize, T kernelSize, T pad, T stride, T dilation, bool ceil_mode) {
     TORCH_CHECK(stride != 0, "stride should not be zero");
+    TORCH_CHECK(pad >= 0,
+                "pad must be non-negative, but got pad: ", pad);
```
Comment (kshitij12345) [inline aten/src/ATen/native/Pool.h:62]: Looking at the issue description, should we also check that `pad <= kernel_size/2` and update the message accordingly?

File: test/test_nn.py [hunk 2]
```diff
@@ -15706,6 +15706,11 @@ def check(expected_out_shape, sizes, *args, **kwargs):
         y = torch.nn.functional.max_pool2d(x, 1, stride=(2, 2), padding=0, ceil_mode=True)
         self.assertEqual(y.size(), (1, 1, 3, 4))
 
+    def test_max_pool1d_neg_idx(self, device):
```
Comment (kshitij12345) [inline test/test_nn.py:15709]: Should this case be moved to OpInfo as error input?

**Review by kshitij12345 — COMMENTED** [review]
@khushi-411 I have a few comments. 

A bigger question that I have is, do we need such checks for `2d` and `3d` version of `maxpool` and also for other pooling ops like `AveragePool` and such.

**Comment by khushi-411** [comment]
> A bigger question that I have is, do we need such checks for 2d and 3d version of maxpool and also for other pooling ops like AveragePool and such.

Hi, @kshitij12345! You are right, I think it will be good to check those and update that. I'm looking into those and will update the PR accordingly. Thanks!

File: aten/src/ATen/native/Pool.h (hunk 1 above)
Comment (khushi-411) [reply aten/src/ATen/native/Pool.h:62]: Yep, added check! Thanks

File: test/test_nn.py (hunk 2 above)
Comment (khushi-411) [reply test/test_nn.py]: Thanks, Kshitij, I've added it :)

I'll add more test cases to ErrorInput in my next commit.

**Comment by khushi-411** [comment]
Hi, Gentle ping, @albanD, @kshitij12345! Could you please look into it once more? I've addressed all the reviews. Thanks!

**Review by albanD — APPROVED** [review]
Thanks for the fix and moving the test to OpInfo!
SGTM

**Comment by khushi-411** [comment]
omg... so many failures

I believe these are base failures and not related to the PR. 😅

Fetching the master branch again to check if these failures are resolved now. Thanks!

**Comment by albanD** [comment]
As you can see on https://hud.pytorch.org/ there are existing breakage on master I'm afraid. So it will still be broken.
If the failure is the same as the one already in master, you can use the `-f` option to bypass them and land this.

@pytorchbot help

**Comment by khushi-411** [comment]
> If the failure is the same as the one already in master, you can use the -f option to bypass them and land this.

Got that! Thanks a lot, @albanD, for looking into it and for reviewing the PR!

**Comment by khushi-411** [comment]
Thanks a lot, @kshitij12345, for reviewing the PR and for merging this! :)
